src/HotChocolate/Core/src/Types/Types/Argument.cs

```diff
@@ -103,6 +103,12 @@ public Argument(ArgumentDefinition definition, int index)
 
         Type = context.GetType<IInputType>(definition.Type!).EnsureInputType();
         _runtimeType = definition.RuntimeType ?? definition.Parameter?.ParameterType!;
+
+        if (definition is DirectiveArgumentDefinition { Property.PropertyType: {} propertyType })
+        {
+            _runtimeType = propertyType;
+        }
```

Comment:
@michaelstaib At first I wasn't sure whether this was the best solution, but after looking deeper into it, I now believe that this is the only (feasible) solution atm.

Reply:
I do not think so :D

Reply:
OK ... reworked this one.